wave_utils/src/config.cpp

```diff
@@ -0,0 +1,362 @@
+#include "wave/utils/config.hpp"
+
+
+namespace wave {
+
+ConfigParser::ConfigParser(void) {
+    this->configured = false;
+    this->loaded = false;
+}
+
+int ConfigParser::getYamlNode(std::string key, YAML::Node &node) {
+    int end;
+    std::string s;
+    std::vector<std::string> elements;
+    std::vector<YAML::Node> traversal;
+
+    // split key into elements
+    s = "";
+    for (int i = 0; i < key.length(); i++) {
+        if (key[i] == '.') {
+            elements.push_back(s);
+            s = "";
+        } else {
+            s += key[i];
+        }
+    }
+    elements.push_back(s);
+
+    // recurse down config key
+    traversal.push_back(this->root);
+    for (int i = 0; i < elements.size(); i++) {
+        end = (traversal.size() - 1);
```

Comment:
Yes. In this specific case,
```cpp
end = (traversal.size() - 1);
node = traversal[end];
```
could become
```cpp
node = traveral.back()
```
and similar in the other place it's used. Just a small readability thing.